src/lib/getheader.rs

```diff
@@ -3,8 +3,8 @@ use pcap::Packet;
 
 // it would seem that this function is working when used with getudp.rs
 // but it does not seem to work in conjunction with gettcp.rs
-pub fn extract_offset_and_ipheader(
-    packet: Packet,
+pub fn extract_offset_and_ipheader<'a>(
```

Comment:
@GyulyVGC 👀 I don't quite understand why this is necessary, could you help be grok this.

Reply:
Hi Dene.
It's important to specify lifetimes in this case because you are also returning a value by reference, so you have to inform the compiler that the returned reference doesn't outlive the function parameter.

Reply:
ah that helps. thanks!